File: packages/java/ygo74-agent-runtime/src/main/java/com/ygo74/agentruntime/domains/endpoints/EndpointAdapters.java
package com.ygo74.agentruntime.domains.endpoints;

import com.ygo74.agentruntime.domains.contracts.ExchangeModels.StandardExchangeRequest;

import java.util.Map;
import java.util.Set;

public final class EndpointAdapters {
    private EndpointAdapters() {}

    private static final Set<String> SUPPORTED = Set.of(
            "openai.chat_completions",
            "openai.responses",
            "anthropic.messages");

    public static StandardExchangeRequest normalize(
            String endpointType,
            String requestId,
            String routeKey,
            Object input,
            boolean stream,
            Map<String, Object> metadata) {

        if (!SUPPORTED.contains(endpointType)) {
            throw new IllegalArgumentException("Unsupported endpoint type: " + endpointType);
        }

        return new StandardExchangeRequest(requestId, routeKey, endpointType, input, stream, metadata, null);
    }
}

File: packages/dotnet/Ygo74.AgentRuntime/Domains/Endpoints/EndpointAdapters.cs
using Ygo74.AgentRuntime.Domains.Contracts;

namespace Ygo74.AgentRuntime.Domains.Endpoints;

public static class EndpointAdapters
{
    private static readonly HashSet<string> Supported =
    [
        "openai.chat_completions",
        "openai.responses",
        "anthropic.messages"
    ];

    public static StandardExchangeRequest Normalize(string endpointType, string requestId, string routeKey, object input, bool stream = false, IDictionary<string, object?>? metadata = null)
    {
        if (!Supported.Contains(endpointType))
        {
            throw new ArgumentException($"Unsupported endpoint type: {endpointType}", nameof(endpointType));
        }

        return new StandardExchangeRequest(requestId, routeKey, endpointType, input, stream, metadata);
    }
}
